jax/test_util.py

```diff
@@ -712,7 +697,15 @@ class JaxTestCase(parameterized.TestCase):
   #   assert core.reset_trace_state()
 
   def setUp(self):
+    super(JaxTestCase, self).setUp()
     core.skip_checks = False
+    # We use the adler32 hash for two reasons.
+    # a) it is deterministic run to run, unlike hash() which is randomized.
+    # b) it returns values in int32 range, which RandomState requires.
+    self._rng = npr.RandomState(zlib.adler32(self._testMethodName.encode()))
```

Comment:
Will this be invariant to the kind of sharding we do on internal tests, or `pytest -n`? That is, will `setUp` be run once before tests are sharded to be run in parallel, with the state somehow getting sequenced between test cases, or will each process run `setUp` itself, meaning the random numbers depend on the sharding amount?

Reply:
`setUp` and `tearDown` run before and after each test case. So yes, this should be independent of any sharding, because it is computed for each test case in isolation.

Reply:
Ah, of course! We want to generate different values for different test cases of the same test. Perfect!